api/logs/v1/labels_enforcer.go

```diff
@@ -197,3 +197,49 @@ func AllowedNamespaces(ctx context.Context) []string {
 	}
 	return namespaces
 }
+
+// WithNamespaceLabelEnforcer returns a middleware that ensures queries do not use both
+// kubernetes_namespace_name and k8s_namespace_name labels.
+func WithNamespaceLabelEnforcer() func(http.Handler) http.Handler {
```

Comment:
Why a new one and not adapting `WithEnforceAuthorizationLabels`? What is the old one not handling well?

Reply:
I opted for creating a new middleware because:
1. these changes are only required for the migration period from ViaQ to OTEL. Then, once we would consider the migration over we could just be able to delete the middleware.
2. minimize the impact on other signals mainly tracing. Since they also use our version of `WithEnforceAuthorizationLabels` 

That being said I'm also happy to port them to `WithEnforceAuthorizationLabels` I would just need to move the check inside both `enforceValuesOnLabelValues` and `enforceValuesOnQuery`

Reply:
I guess it is ok for a migration-period-only middleware. Suggest to name it like this to make this visible for the code reader in 3/5 mo time.